bin/dadmin

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+set -e
+
+function cleanup {
+  # capture exit code
+  code=$?
+  echo "cleaning up"
+
+  # ignore errors
+  set +e
+  docker-compose down
+
+  exit $code
+}
+
+trap cleanup EXIT
```

Comment:
I don't recommend having this boilerplate taken from the other scripts. It will shut everything down any time you run this script. Probably not what you want to do.

Something like this should be sufficient:
```sh
#!/bin/sh

if [ $# -gt 0 ]
then
  docker-compose run --rm web rails runner "Member.find_by(name: '$@').add_role(:admin)"
else
  docker-compose run --rm web rails runner "Member.last.add_role(:admin)"
fi
```

Reply:
OOOkay, I get it. Doh! Thanks biggianteye. The start of the original script clears up the current docker instance? Marvelous.

Reply:
This is a tangent, but there are two things happening.

1. `trap cleanup EXIT`

The  `trap` keyword allows you to intercept various signals received by the script. This particular line means that when the `EXIT` signal is received (ie. normal termination), the `cleanup` function is called.

This minimal example demonstrates this behaviour:
```sh
#!/bin/sh

function cleanup
{
    echo "Cleanup function being called"
}

trap cleanup EXIT

echo "Hello world"
```

```
[burhan@enceladus ~] {0} $ ./test.sh
Hello world
Cleanup function being called
```

2. `docker-compose down`

This is the important line in the `cleanup` function. It tells docker to bring down all the containers defined in `docker-compose.yml`. ie. `web` (the web server) and `db` (the database).

Reply:
Thanks that's really clear. I've only recently worked out how important scripts are for getting work done. Marvellous, a 3 person solution.

Reply:
Cool, thanks! I made suggested changes to a script and now it's much better.